### PR TITLE
Batch Normalization should squeeze mean/var/beta/gama tensors when calling tf.nn.fused_batch_norm

### PR DESCRIPTION
**Pull request by yanboliang**
### Summary
```BatchNormalization``` layer, mean/var/beta/gama tensors may be [broadcast](https://github.com/keras-team/keras/blob/master/keras/layers/normalization.py#L140). 
```
input = Input(shape=(3, 1001, 1001), dtype='float32')
x = Conv2D(filters=64, kernel_size=(3, 3), strides=1, padding='same')(input)
x = BatchNormalization(axis=1)(x)
```
In the above example, as you set ```axis=1```, the mean/var/beta/gama tensors would be reshaped as [1,64,1,1]. For TF backend, it will try to call the optimized ```tf.nn.fused_batch_norm``` if possible. But ```tf.nn.fused_batch_norm``` only accept mean/var/beta/gama as 1 dimension tensor, we need to squeeze them ahead.
Actually we have did same preprocessing for [CNTK backend](https://github.com/keras-team/keras/blob/master/keras/backend/cntk_backend.py#L1070).

Note: This bug should only happened when TF-GPU as backend, since TF-CPU FusedBatchNorm doesn't support NCHW format, and it will fall back to the default ```tf.nn.batch_normalization```.   

### Related Issues
https://github.com/keras-team/keras/issues/10648

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by samiatwork** [comment]
Is there a workaround for this?

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -1879,10 +1879,18 @@ def batch_normalization(x, mean, var, beta, gamma, axis=-1, epsilon=1e-3):
             tf_data_format = None
 
         if tf_data_format == 'NHWC' or tf_data_format == 'NCHW' and _has_nchw_support():
+            # The mean / var / beta / gamma may be processed by broadcast
+            # so it may have extra axes with 1, it is not needed and should be removed
+            mean = tf.squeeze(mean)
+            var = tf.squeeze(var)
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Only call `squeeze` if `mean` / `var` have more than 1 axes.

**Comment by yanboliang** [comment]
@fchollet It's not necessary to add new unit test, actually the [existing one](https://github.com/keras-team/keras/blob/master/tests/keras/layers/normalization_test.py#L39) already cover this case. As Travis CI does not run on GPU, so it passed. I have run ```pytest ./tests/keras/layers/normalization_test.py``` on a GPU node with TF-GPU installed, the master code doesn't pass(same error as mentioned in this PR), and passed after this fix. Thanks.

**Comment by HAPPIER-Music** [comment]
This works-->https://github.com/keras-team/keras/issues/10382#issuecomment-400728875
